### PR TITLE
[Release] Update release process documentation

### PR DESCRIPTION
**Pull request by palatter**
# What

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: docs/releases/creating-a-release.md [hunk 1]
```diff
@@ -13,6 +13,20 @@ _This document applies to beta and stable releases. Alpha releases are created n
...
+
+Monday - 
+Tuesday - 
+Wednesday - 
+Thursday - 
+Friday - 
+
 ## Step 0: Preparation
 
 ### Step 0.1: Update Strings on Main
```
Comment (JamesBurnside) [inline docs/releases/creating-a-release.md:32]: > String translation can take up to 5 working days to complete. **This needs to be done the week before release.**

This is a little confusing. We send strings for translation after every PR. What this needs to be is something like:

```md
- String translation take up to 5 working days to complete. **Any PR changing strings needs to be complete 5 working days before release.**
```

Comment (JamesBurnside) [reply docs/releases/creating-a-release.md:32]: (I know not part of this PR, but just making notes as I run release this week)

File: docs/releases/creating-a-release.md [hunk 2]
```diff
@@ -80,6 +107,10 @@ graph LR
   pr -.-o|Create Pull Request| prerelease
 ```
 
+### Step 1.2: Groom changelog
```
Comment (JamesBurnside) [inline docs/releases/creating-a-release.md:110]: This and the next step are both `1.2`

**Review by dmceachernmsft — APPROVED** [review]

**Review by jpeng-ms — APPROVED** [review]
Thanks for updating it!

**Pull request merged**